xbmc/platform/darwin/osx/XBMCApplication.mm

```diff
@@ -420,25 +429,20 @@ int main(int argc, char* argv[])
     // Ensure the application object is initialised
     [NSApplication sharedApplication];
 
-    CPSProcessSerNum PSN;
-    /* Tell the dock about us */
-    if (!CPSGetCurrentProcess(&PSN))
-      if (!CPSEnableForegroundOperation(&PSN, 0x03, 0x3C, 0x2C, 0x1103))
-        if (!CPSSetFrontProcess(&PSN))
-          [NSApplication sharedApplication];
-
+    // Make sure we call applicationWillTerminate on SIGTERM
+    [NSProcessInfo.processInfo disableSuddenTermination];
 
     // Set up the menubars
     [NSApplication.sharedApplication setMainMenu:[[NSMenu alloc] init]];
     setupApplicationMenu();
     setupWindowMenu();
 
-    // Create XBMCDelegate and make it the app delegate
-    xbmc_delegate = [[XBMCDelegate alloc] init];
-    [NSApplication.sharedApplication setDelegate:xbmc_delegate];
-
+    // Set App Delegate
+    [NSApp setDelegate:[XBMCDelegate new]];
```

Comment:
delegate will die after this line because the property is `weak`.

```suggestion
    auto appDelegate = [XBMCDelegate new];
    NSApp.delegate = appDelegate;
```